src/main.rs

```diff
@@ -1,35 +1,690 @@
+/// ```
+/// fn open_gift_or_panic<T>(gift: GiftBox<T>) -> T {
+///      return match gift {
+///         GiftBox::Gifts(the_gift) => the_gift,
+///         GiftBox::Empty => panic!("I thought I was going to get a gift! D: "),
+///     }
+/// }
+/// ```
+/// Here, the function works for any `T` because the code works regardless of what the type in the
+/// gift is, it only needs to return the type which is whatever type was in the gift box. In the
+/// case of the generic function, it passes the responsibility of figuring out what the `T` is on to
+/// the caller of the function.
+///
+#[derive(Copy, Debug, PartialEq, PartialOrd)]
+pub enum GiftBox<T> {
     Gifts(T),
     Empty
 }
 
-/// In this example an empty gift box is declared and no gift box is declared with gifts. This is
-/// doomed to compilation failure because the compiler does not know what type the gifts could be.
-fn main() {
-    // Declare an empty gift box called `empty_box`.
-    let empty_box = GiftBox::Empty;
+impl<T> GiftBox<T> {
+    /// The `fill(t: Option<T>)` method accepts an [`std::option::Option`] and returns a `GiftBox`
+    /// with either `GiftBox::Gifts(t)` if `Some(t)` was provided or a `GiftBox::Empty` if
+    /// `None` is provided.
+    ///
+    /// #Arguments
+    /// * `t: Option<T>` - `t` accepts an [`std::option::Option`]
+    ///
+    /// #Returns
+    /// Returns a `GiftBox<T>`.
+    ///
+    /// #Example
+    /// Filling a `GiftBox` with `Some(T)`:
+    /// ```
+    /// let filled_box = GiftBox::fill(Some(["Toys", "Candy", "Money"]));
+    /// assert_eq!(filled_box, GiftBox::Gifts(["Toys", "Candy", "Money"]));
+    /// ```
+    /// Filling a `GiftBox` with `None`:
+    /// ```
+    /// let empty_box: GiftBox<()> = GiftBox::fill(None);
+    ///  assert_eq!(empty_box, GiftBox::Empty);
+    /// ```
+    pub fn fill(t: Option<T>) -> GiftBox<T> {
+        match t {
+            Some(t) => GiftBox::Gifts(t),
+            None => GiftBox::Empty,
+        }
+    }
+
+    /// The `fill_keeping_some(t: Option<T>)` method accepts an
+    /// [`std::option::Option`] and returns a `GiftBox` with either `GiftBox::Gifts(Some(t))` if
+    /// `Some(t)` was provided or a `GiftBox::Empty` if `None` is provided.
+    ///
+    /// #Arguments
+    /// * `t: Option<T>` - `t` accepts an [`std::option::Option`]
+    ///
+    /// #Returns
+    /// Returns a `GiftBox<Option<T>>`.
+    ///
+    /// #Example
+    /// Filling a `GiftBox` with `Some(T)`:
+    /// ```
+    /// let filled_box = GiftBox::fill_keeping_some(Some(["Toys", "Candy", "Money"]));
+    /// assert_eq!(filled_box, GiftBox::Gifts(Some(["Toys", "Candy", "Money"])));
+    /// ```
+    /// Filling a `GiftBox` with `None`:
+    /// ```
+    /// let empty_box: GiftBox<()> = GiftBox::fill_keeping_some(None);
+    ///  assert_eq!(empty_box, GiftBox::Empty);
+    /// ```
+    pub fn fill_keeping_some(t: Option<T>) -> GiftBox<Option<T>>{
```

Comment:
This function isnt necessary, as the existing "fill" function can be used. 
```rust
// No need for this
let filled_box = GiftBox::fill_keeping_some(Some(["Toys", "Candy", "Money"]));
// When you can do this
let filled_box = GiftBox::fill(Some(Some(["Toys", "Candy", "Money"])));
// Simplified top version
let my_gift = Some(["Toys", "Candy", "Money"]);
let filled_box = GiftBox::fill_keeping_some(my_gift);
// Simplified bottom version
let my_gift = Some(["Toys", "Candy", "Money"]);
let filled_box = GiftBox::fill(Some(my_gift));
```
The key here is that since T can be anything, that includes being an Option itself as well. So if you want to "keeping_some" then that just means the type of your gift is just an option, and you can use the normal fill function while passing Option<String[3]> as the gift type.

Reply:
This function was removed in https://github.com/hypnoseal/giftbox/commit/a285af5667db26a82367d80711c283d3bcfa3d1b